packages/generator/src/service-vdm/v4/complex-type-vdm.ts

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { transformComplexTypesBase } from '../common/complex-type-vdm';
+import { ServiceNameFormatter } from '../../service-name-formatter';
+import { VdmComplexType } from '../vdm-types';
+import { ServiceMetadata } from '../../parser/util/edmx-types';
+import { EdmxComplexType } from '../../parser/v4/edmx-types';
+import { parseComplexTypes } from '../../parser/v4/edmx-parser';
+
+export function joinComplexTypes(
+  complexType: EdmxComplexType,
+  baseType: EdmxComplexType
+): EdmxComplexType {
+  return {
+    ...complexType,
+    Property: [...complexType.Property, ...baseType.Property]
+  };
+}
+
+export function getComplexTypesV4(
```

Comment:
The parsing of the complex types in the edmx is different. For v4  the joinTypesWithBaseTypes is executed. I could of course hand over the edmx type to the transform method and remove the two implementations.

Reply:
but then it would so be so symmetric in the generation step: 
```
  const functionImportsWithoutReturnType = isV2Metadata(serviceMetadata.edmx)
      ? generateFunctionImportsV2(serviceMetadata, formatter)
      : generateFunctionImportsV4(serviceMetadata, formatter);

    const complexTypes = isV2Metadata(serviceMetadata.edmx)
      ? generateComplexTypesV2(serviceMetadata, formatter)
      : generateComplexTypesV4(serviceMetadata, formatter);

    const entities = isV2Metadata(serviceMetadata.edmx)
      ? generateEntitiesV2(serviceMetadata, complexTypes, formatter)
      : generateEntitiesV4(serviceMetadata, complexTypes, formatter);
```